src/Registry.php

```diff
@@ -827,4 +840,104 @@ protected function toFlatten($separator = null, $data = null, array &$array = []
             $array[$key] = $v;
         }
     }
+
+    /**
+     * Magic method to access separator property.
+     *
+     * @param  string  $name  The name of the property.
+     *
+     * @return string|null A value if the property name is valid, null otherwise.
+     *
+     * @since       __DEPLOY_VERSION__
+     * @deprecated  3.0  This is a B/C proxy for deprecated read accesses
+     */
+    public function __get($name)
+    {
+        switch ($name) {
+            case 'separator':
+                \trigger_deprecation(
```

Comment:
According to this https://3v4l.org/fgOsF users will never see this deprecation notice.